server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1645,7 +1645,10 @@ void getAssignments(SortedMap<TServerInstance,TabletServerStatus> currentStatus,
     AssignmentParamsImpl params =
         AssignmentParamsImpl.fromThrift(currentStatus, currentTServerGroups,
             unassigned.entrySet().stream().collect(HashMap::new,
-                (m, e) -> m.put(e.getKey(), e.getValue().getServerInstance()), Map::putAll),
```

Comment:
Was this an existing NPE bug?

Reply:
Honestly, I don't remember. Looking at the code though, it was using the value of the current location, or null. Instead of the last location or null.